lib/fed_email_domain_downloader.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'faraday'
+require 'pry'
+require 'fileutils'
+
+class FedEmailDomainDownloader
+  attr_reader :destination
+
+  DOT_GOV_DOWNLOAD_PATH = 'https://raw.githubusercontent.com/cisagov/dotgov-data/main/current-federal.csv'
+
+  def initialize(destination: 'tmp/fed_download_path')
+    @destination = destination
+  end
+
+  def dot_gov_csv_path
+    response = Faraday.get(DOT_GOV_DOWNLOAD_PATH)
+    response.body
+  end
+
+  def run!
+    FileUtils.mkdir_p(destination)
+    csv ||= CSV.parse(dot_gov_csv_path, col_sep: ',', headers: true)
```

Comment:
I don't think the memoization saves anything here this is a local variable
```suggestion
    csv = CSV.parse(dot_gov_csv_path, col_sep: ',', headers: true)
```